docs/development/mh-rest-api-v1-vs-v2.md

```diff
@@ -0,0 +1,136 @@
+---
+layout: default
+title: Verschillen tussen REST API v1 en v2
+has_children: false
+has_toc: true
+nav_order: 3
+last_modified_date: 2021-04-13T06:33:08+0100
+---
+
+<details markdown="block">
+  <summary>
+    Inhoudstafel
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+Status: WIP
+{: .label .label-yellow }
+
+# REST API v1 vs v2
+
+MediaHaven biedt sinds het begin een REST API aan om bewerkingen uit te kunnen
+voeren op mediaobjecten. Vanaf MediaHaven versie 21.1 (uitgebracht op 16 april
+2021) werd van deze REST API een nieuwe versie uitgerold. We zullen verder deze
+twee versies van elkaar onderscheiden met de aanduidingen "v1" en "v2".
+
+Hoewel veel van de basisfuncties in de nieuwe REST API (zoals het opvragen
+van een _resource_, het uitvoeren van een zoekopdracht,...) quasi onveranderd
+zijn gebleven, werden er ook wijzigingen ingevoerd die _backwards incompatible_
+zijn. Hierdoor zouden bestaande integraties die gebruik maken van de REST API
+v1 kunnen breken wanneer ze naar het nieuwe _endpoint_ worden overgezet.
+
+De oude REST API v1, op het huidige _endpoint_, wordt nog ondersteund tot
+**MediaHaven versie 22.1**, maar we raden sterk aan om naar de nieuwe REST
+API v2, op het nieuwe _endpoint_, over te stappen.
+
+## Endpoints
+
+Zeticon koos voor het versiebeheer van hun REST API voor versiebeheer via het
+URL-pad. Dit betekent dat het versienummer toegevoegd wordt aan het URL-pad om
+de v2 van de REST API te bereiken. Concreet:
+
+| API versie | Endpoint                                        |
+|-----------:|-------------------------------------------------|
+| v1         | https://archief.viaa.be/mediahaven-rest-api/    |
+| v2         | https://archief.viaa.be/mediahaven-rest-api/v2/ |
+
+**Note**: Dit zijn de _endpoints_ voor de productieomgeving! Voor een
+beschrijving van de verschillende omgevingen en hun _endpoints_, zie [Alles om te
+starten met het bouwen van een integratie]({{ site.baseurl }}{% link
+docs/development/index.md %}).
+
+## Verschillen tussen REST API v1 en v2
+
+Hieronder lijsten we eerst kort de belangrijkste _backwards incompatible_
+verschillen op tussen de REST API v1 en v2. Daarna gaan we dieper in op wat
+dat concreet inhoudt.
+
+- authenticatie kan énkel nog via OAuth2: _Basic Auth_ wordt niet meer ondersteund
+- de structuur van de JSON _response_ wijzigt
+- de _resource_ `/media` is nu `/records` geworden
+
```

Comment:
En `resources` in het pad is weg:
v1: `https://host/mediahaven-rest-api/resources/roles`
v2: `https://host/mediahaven-rest-api/v2/roles/`